File: src/api/episodes/episode.model.js
//
// ─── EPISODE MODEL ──────────────────────────────────────────────────────────────
//

const mongoose = require('mongoose');

const { Schema } = mongoose; 

const EpisodeSchema = new Schema({
  episodeName: { type: String, required: true },
  episodeNumber: { type: Number, min: 1 , max: 30, required: true },
  seasonNumber: { type: Number, min: 1, max: 25, required: true },
  characterId: [{ type: Schema.Types.ObjectId, ref: 'Character' }],
  lineId: [{ type: Schema.Types.ObjectId, ref: 'Line' }],
});

EpisodeSchema.pre('find', function (next) {
  this.populate('lineId')
  next()
})

EpisodeSchema.pre('findOne', function (next) {
  this.populate('lineId')
  next()
});

const Episode = mongoose.model('Episode', EpisodeSchema); 

module.exports = {
  Episode,
};

File: models/episode.model.go
package models

type Episode struct {
  ID            bson.ObjectId     `json:"_id,omitempty" bson:"_id,omitempty"`
  EpisodeName   string            `json:"episodeName,omitempty" bson:"episodeName,omitempty"`
  // TODO: FINISH
  EpisodeNumber int               `min: "1", max: "30", required: "true"`
	SeasonNumber  int               `min: "1", max: "25", required: "true"`
  CharacterId   []bson.ObjectId   'bson:"_id,omitempty" json:"object'
  LineId:       []bson.ObjectId   [{ type: Schema.Types.ObjectId, ref: 'Line' }],
}
